hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyReadWriteOps.java

```diff
@@ -0,0 +1,252 @@
+
+  /**
+   * Task type of read task, or write task.
+   */
+  public enum TaskType {
+    READ_TASK,
+    WRITE_TASK
+  }
+  private KeyGeneratorUtil kg;
+
+
+  @Override
+  public Void call() throws Exception {
+    init();
+    OzoneConfiguration ozoneConfiguration = createOzoneConfiguration();
+    OzoneClient[] ozoneClients = new OzoneClient[clientsCount];
+    for (int i = 0; i < clientsCount; i++) {
+      ozoneClients[i] = createOzoneClient(omServiceID, ozoneConfiguration);
+    }
+
+    ensureVolumeAndBucketExist(ozoneClients[0], volumeName, bucketName);
+    ozoneBuckets = new OzoneBucket[clientsCount];
+    for (int i = 0; i < clientsCount; i++) {
+      ozoneBuckets[i] = ozoneClients[i].getObjectStore().getVolume(volumeName)
+              .getBucket(bucketName);
+    }
+
+    timer = getMetrics().timer("key-read-write");
+    if (objectSizeInBytes >= 0) {
+      keyContent = RandomUtils.nextBytes(objectSizeInBytes);
+    }
+    if (kg == null) {
+      kg = new KeyGeneratorUtil();
+    }
+    runTests(this::readWriteKeys);
+
+    for (int i = 0; i < clientsCount; i++) {
+      if (ozoneClients[i] != null) {
+        ozoneClients[i].close();
+      }
+    }
+    return null;
+  }
+
+  public void readWriteKeys(long counter) throws Exception {
+    int clientIndex = (int)((counter) % clientsCount);
+    OzoneBucket ozoneBucket = ozoneBuckets[clientIndex];
+    TaskType taskType = decideReadOrWriteTask();
+    String keyName = getKeyName(clientIndex);
+
+    timer.time(() -> {
+      try {
+        switch (taskType) {
+        case READ_TASK:
+          processReadTasks(keyName, ozoneBucket);
+          break;
+        case WRITE_TASK:
+          processWriteTasks(keyName, ozoneBucket);
+          break;
+        default:
+          break;
+        }
+      } catch (RuntimeException ex) {
+        LOG.error(ex.getMessage());
+        throw ex;
+      } catch (IOException ex) {
+        LOG.error(ex.getMessage());
+        throw new RuntimeException(ex.getMessage());
+      }
+
+    });
+  }
+
+  public void processReadTasks(String keyName, OzoneBucket ozoneBucket)
+          throws RuntimeException, IOException {
+    if (readMetadataOnly) {
+      ozoneBucket.getKey(keyName);
+    } else {
+      byte[] data = new byte[objectSizeInBytes];
+      try (OzoneInputStream inputStream = ozoneBucket.readKey(keyName)) {
+        inputStream.read(data);
+      }
+    }
+  }
+  public void processWriteTasks(String keyName, OzoneBucket ozoneBucket)
+          throws RuntimeException, IOException {
+    try (OzoneOutputStream out =
+                 ozoneBucket.createKey(keyName, objectSizeInBytes)) {
+      out.write(keyContent);
+    } catch (Exception ex) {
+      throw ex;
+    }
+  }
+  public TaskType decideReadOrWriteTask() {
+    if (!isMixWorkload()) {
+      if (percentageRead == 100) {
+        return TaskType.READ_TASK;
+      } else {
+        return TaskType.WRITE_TASK;
+      }
+    }
```

Comment:
Maybe it's simpler this way:

```suggestion
    if (percentageRead == 100)) {
      return TaskType.READ_TASK;
    } else if (percentageRead == 0) {
      return TaskType.WRITE_TASK;
    }
```

Reply:
oh!! this definitely make sense!! cool